jetson/nav/recovery.hpp

```diff
@@ -0,0 +1,104 @@
+#include <chrono>
+#include <cmath>
+#include <ctime>
+#include <deque>
+#include <memory>
+
+#include <eigen3/Eigen/Core>
+
+#include "rover.hpp"
+#include "rover_msgs/Odometry.hpp"
+
+using Eigen::Vector2d;
+using namespace rover_msgs;
+using time_point = std::chrono::high_resolution_clock::time_point;
+
+struct OdomTimePoint {
+    Odometry odom;
+    time_point time;
+};
+
+struct RecoveryPoint {
+    Odometry odom;
+    // if rover should drive backwards to point
+    bool backwards;
+};
+
+
+/***
+ * Recovery Watchdog
+ * Class used to detect if Rover has become stuck
+ * and generate recovery manuevers to get unstuck.
+ */
+class Recovery {
+public:
+    Recovery(std::chrono::duration<int> offsetAmount, double dispThresh,
+             std::chrono::duration<int> maxTurnTime);
```

Comment:
Add a `using` alias for `std::chrono::duration<int>` since that is a mouth full

https://www.nextptr.com/tutorial/ta1193988140/how-cplusplus-using-or-aliasdeclaration-is-better-than-typedef

Reply:
Can use search and replace in editor